### PR TITLE
add warning if target path is . ./. ./.. or ..

### PR DESCRIPTION
**Pull request by zuzuleinen**
don't allow paths that are . ./. ./.. or ..

### CODE REVIEW

File: bobtask/verify.go [hunk 1]
```diff
@@ -32,9 +35,36 @@ func (t *Task) verifyBefore() (err error) {
...
+				return usererror.Wrap(fmt.Errorf("invalid target `%s` for task `%s`", path, t.name))
+			}
+		}
+	}
+
 	return nil
 }
 
+// verifyTargetPath reports whether the final component of path is "." or ".."
+func verifyTargetPath(path string) bool {
```
Comment (Equanox) [inline bobtask/verify.go:38]: The function definition/usage is a bit misleading. If `verifyTargetPath() returns true we should not expect to throw an error. Maybe rename to isValidFilesystemTarget() and adjust it's behavior.

Also: Does it make sense to only validate the last part of the path? Shouldn't it check each subpart of the path?
I can imagine targets like those:
* my/../strange/target
* ../target

Would also be nice to have a unit test for this function.

**Review by Equanox — CHANGES_REQUESTED** [review]

File: bobtask/verify.go (hunk 1 above)
Comment (zuzuleinen) [reply bobtask/verify.go]: Updated the tests to cover also /../ parts

File: bobtask/verify.go [hunk 2]
```diff
@@ -16,9 +24,35 @@ func (t *Task) VerifyAfter() error {
...
+			if !isValidFilesystemTarget(path) {
+				return usererror.Wrap(fmt.Errorf("invalid target `%s` for task `%s`", path, t.name))
+			}
+		}
+	}
+
 	return nil
 }
 
+// isValidFilesystemTarget reports whether the final component of path is "." or ".."
```
Comment (Equanox) [inline bobtask/verify.go:38]: Comment is missleading

Comment (zuzuleinen) [reply bobtask/verify.go]: Will update

File: bobtask/verify.go [hunk 3]
```diff
@@ -16,9 +24,36 @@ func (t *Task) VerifyAfter() error {
...
 
+// isValidFilesystemTarget checks if target is a valid path
+// paths which resolve to `.` or containing `..` are considered invalid
+func isValidFilesystemTarget(path string) bool {
+	// check for end with one dot
+	if path == "." {
+		return false
+	}
+
+	if strings.Contains(path, "..") {
```
Comment (meain) [inline bobtask/verify.go:46]: Might not be common, but something like `file..extension` should be valid.

Comment (Equanox) [reply bobtask/verify.go]: Thx for mentioning it. No leveraging `filepath.Clean()` which should handle those cases correctly.

**Review by Equanox — APPROVED** [review]

**Pull request merged**